### PR TITLE
Fixes a few NPEs when unboxing null Integer

### LINKED ISSUE #443: Crashing !!!
09-29 16:46:25.638 27959 27959 E AndroidRuntime: java.lang.NullPointerException: Attempt to invoke virtual method 'int java.lang.Integer.intValue()' on a null object reference
09-29 16:46:25.638 27959 27959 E AndroidRuntime:        at com.facebook.stetho.inspector.elements.Document$4.store(Document.java:394)
09-29 16:46:25.638 27959 27959 E AndroidRuntime:        at com.facebook.stetho.inspector.elements.ShadowDocument$Update.getChangedElements(ShadowDocument.java:297)
09-29 16:46:25.638 27959 27959 E AndroidRuntime:        at com.facebook.stetho.inspector.elements.Document.applyDocumentUpdate(Document.java:378)
09-29 16:46:25.638 27959 27959 E AndroidRuntime:        at com.facebook.stetho.inspector.elements.Document.updateTree(Document.java:333)
09-29 16:46:25.638 27959 27959 E AndroidRuntime:        at com.facebook.stetho.inspector.elements.Document.access$1200(Document.java:30)
09-29 16:46:25.638 27959 27959 E AndroidRuntime:        at com.facebook.stetho.inspector.elements.Document$ProviderListener.onPossiblyChanged(Document.java:688)
09-29 16:46:25.638 27959 27959 E AndroidRuntime:        at com.facebook.stetho.inspector.elements.android.AndroidDocumentProvider$1.run(AndroidDocumentProvider.java:67)
09-29 16:46:25.638 27959 27959 E AndroidRuntime:        at android.os.Handler.handleCallback(Handler.java:739)
09-29 16:46:25.638 27959 27959 E AndroidRuntime:        at android.os.Handler.dispatchMessage(Handler.java:95)
09-29 16:46:25.638 27959 27959 E AndroidRuntime:        at android.os.Looper.loop(Looper.java:148)
09-29 16:46:25.638 27959 27959 E AndroidRuntime:        at android.app.ActivityThread.main(ActivityThread.java:5417)
09-29 16:46:25.638 27959 27959 E AndroidRuntime:        at java.lang.reflect.Method.invoke(Native Method)
09-29 16:46:25.638 27959 27959 E AndroidRuntime:        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
09-29 16:46:25.638 27959 27959 E AndroidRuntime:        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)

### PR DESCRIPTION
**Pull request by felipecsl**
Fixes #443

### CODE REVIEW

**Comment by longinoa** [comment]
Thanks @felipecsl!

Im not super familiar with this code, do we also need to check the ternary operators in `ChildEventingList#acquire` and `ChildEventingList#addWithEvent` It seems like we would need to.

Also the code in `ChildEventingList` seems to imply that parentId should be -1 when null should we be translating this to -1 rather than not calling `onChildNodeRemoved`?

cc @rickbrew

**Comment by felipecsl** [comment]
@longinoa yeah there's a couple other places where I wasn't sure what to do. I'd be happy to update the ternary operators to handle null too

**Comment by rickbrew** [comment]
Do we have a reliable (and hopefully minimal) repro for this? I'm afraid this is fixing a symptom, but not the actual cause, and will result in incorrect or incoherent state in the elements tree. This code was very carefully constructed, which leads me to believe that using `int` instead of `Integer` is intentional (a cheap way to do an assert, in other words).

Let me spend some time to (finally) investigate these code paths. A repro would be tremendously helpful though so I can catch it in the act.

File: build.gradle [hunk 1]
```diff
@@ -3,7 +3,7 @@ buildscript {
         jcenter()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:2.1.3'
+        classpath 'com.android.tools.build:gradle:2.2.2'
```
Comment (rickbrew) [inline build.gradle]: why the version bump here?

Comment (felipecsl) [inline build.gradle]: this was added automatically by Android Studio, but i'd be happy to revert

**Comment by longinoa** [comment]
@felipecsl see the latest comment on #381 do you have an easy repro?

**Comment by felipecsl** [comment]
@longinoa I can really easily reproduce on our app by just navigating around, but I'm not sure what exactly triggers it.

**Comment by longinoa** [comment]
@felipecsl it would help @rickbrew to have a local repro, is your app open source?

**Comment by felipecsl** [comment]
Unfortunately not. Maybe @pollux has a repro case?

**Comment by rickbrew** [comment]
For instance, it's unclear from local code inspection whether 1) `oldElementInfo.parentElement` is `null`, or 2) it's non-`null` but missing from `mObjectIdMapper`. Is `getGarbageElements` pushing elements in the wrong order? Is the application itself violating some invariant that I incorrectly thought would hold true? etc.

Even if we have a fix, I'll still need a repro in order to really validate that the algorithm and data structures for this are good to go.

File: build.gradle (hunk 1 above)
Comment (rickbrew) [inline build.gradle]: okie dokie 👍

**Comment by felipecsl** [comment]
I agree that blindly fixing by doing defensive checks is bad. In our case, the `Integer` returned from `getIdForObject()` was null

**Comment by felipecsl** [comment]
We're using React Native, if that matters. It seems like there could be a correlation between that and the crash

**Comment by rickbrew** [comment]
So you have your own (non-minimal) repro? What is `oldElementInfo.parentElement` ? Is it a `Fragment` with `setRetainInstance(true)` ?

**Comment by longinoa** [comment]
See #459 Thanks to @Axolot1 looks like a repro

**Comment by felipecsl** [comment]
OK that's great!

**Comment by longinoa** [comment]
Hey @felipecsl Thanks for the PR but there was a larger logic issue in play. See the commit https://github.com/facebook/stetho/commit/37e615ed293ab49073f57b37e51fdcc6334cf919

Stetho has been updated to 1.4.2 to address this issue.

**Pull request closed without merging**